### PR TITLE
Polishes cli 

### PR DESCRIPTION
**Pull request by sr-gi**
The `cli` was a bit messy. All functions returned Value/`None` what made it hard to test and review.

Performs a major refactor to switch to single return + exceptions, reviews comments + docstrings and makes the code more modular.

### CODE REVIEW

File: cli/README.md [hunk 1]
```diff
@@ -21,15 +21,26 @@ Refer to [INSTALL.md](INSTALL.md)
...
+The command line interface has, currently, four commands:
 
-- `add_appointment`: registers a json formatted appointment to the tower.
+- `register`: registers your user with the tower.
+- `add_appointment`: sends a json formatted appointment to the tower.
 - `get_appointment`: gets json formatted data about an appointment from the tower.
 - `help`: shows a list of commands or help for a specific command.
 
+### register
+This commands serves as registration. It sends your public key to the tower to create a subscription (fee atm) and returns a number of available appointment slots in the tower. Toping up the subscription can be done by simply sending a register message again.
```
Comment (bigspider) [inline cli/README.md:32]: typo "topping up".

File: cli/README.md [hunk 2]
```diff
@@ -21,15 +21,26 @@ Refer to [INSTALL.md](INSTALL.md)
...
+
+Notice that you need to be register before sending any other type of request to the tower.
```
Comment (bigspider) [inline cli/README.md:34]: "need to register" or "need to be registered"

File: test/teos/e2e/test_basic_e2e.py [hunk 3]
```diff
@@ -266,7 +272,7 @@ def test_two_identical_appointments(bitcoin_cli, create_txs):
 
 
 def test_two_appointment_same_locator_different_penalty_different_users(bitcoin_cli, create_txs):
-    # This tests sending an appointment with two valid transaction with the same locator.
+    # This tests sending an appointment with two valid transaction with the same locator fro different users
```
Comment (orbitalturtle) [inline test/teos/e2e/test_basic_e2e.py:275]: Typo - from*

Comment (orbitalturtle) [reply test/teos/e2e/test_basic_e2e.py:275]: or for*?

**Pull request merged**